src/gitchangelog/gitchangelog.py

```diff
@@ -1349,6 +1355,93 @@ def render_commit(commit, opts=opts):
             yield render_version(version) + "\n\n"
 
 
+@available_in_config
+def kolibree_output(data, opts={}):
+    """Returns Markdown Text changelog content from data
+
+    Connect to GitHub if commit is missing a body and retrieve PR description.
+    """
+
+    token = opts.get("github_token")
+    repo = opts.get("github_repo")
+    if not all([token, repo]):
+        die("Github token, repo and/or PR regexp config is missing.")
+
+    # GitHub
+    github = None
+    if Github:
+        try:
+            github = Github(token)
+            repo = github.get_repo(repo)
+        except Exception:
+            die("Unable to connect to Github")
+
+        re_pr_num = re.compile(
+            # Example: "Title of commit (#PR_NUM)"
+            r"""
+                \(
+                \#(\d+)     # wrapped in parentheses for direct number match
+                \)
+                $
+            """,
+            re.X,
+        )
+
+    def rest_title(label, char="="):
+        return (label.strip() + "\n") + (char * len(label) + "\n")
+
+    def render_version(version):
+        title = "%s (%s)" % (version["tag"], version["date"]) \
+                if version["tag"] else \
+                opts["unreleased_version_label"]
+        s = rest_title(title, char="-")
+
+        sections = version["sections"]
+        nb_sections = len(sections)
+        for section in sections:
+
+            section_label = section["label"] if section.get("label", None) \
+                            else "Other"
+
+            if not (section_label == "Other" and nb_sections == 1):
+                s += "\n" + rest_title(section_label, "~")
+
+            for commit in section["commits"]:
+                s += render_commit(commit)
+        return s
+
+    def render_commit(commit, opts=opts):
+        subject = commit["subject"]
+
+        subject += " [%s]" % (", ".join(commit["authors"]), )
+
+        entry = indent('\n'.join(textwrap.wrap(subject)),
+                       first="- ").strip() + "\n"
+
+        if commit["body"]:
+            entry += "\n" + indent(commit["body"]) + "\n"
+        else:
+            # Get GitHub PR description/body
+            pr_num = re_pr_num.search(commit["subject"])
+            pr_num = pr_num.groups()[0] if pr_num else None
+            if pr_num:
+                try:
+                    pull = repo.get_pull(int(pr_num))
+                    entry += "\n```\n" + pull.body + "\n```\n"
```

Comment:
Yes it can be quite large. Not sure how we wanna handle it.
Perhaps with additional regex which will parse pull request template and extract only description section?

Reply:
If it can be done easily with a regex then that sounds like a good solution.

Reply:
Added regexp which extracts and fills changelog entry body with only PR description text (discarding all other PR template info)